### PR TITLE
feat: add `OnyxLink` component

### PR DESCRIPTION
**Pull request by larsrickert**
relates to #237

Add a default `OnyxLink` component. States like hover, active etc. will be implemented in a follow-up PR once they are clarified with UX.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] All changes are documented in the documentation app (folder `apps/docs`)
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxLink/types.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   */
+  target?: LinkTarget;
+  /**
+   * Text size.
+   */
+  size?: TextSize;
+  /**
+   * If `true`, the monospace font family will be used instead of the default one.
+   */
+  monospace?: boolean;
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxLink/types.ts:20]: We shouldn't use properties for styling. The component will mainly be used inline in running text, where the font styling should come from the surrounding text. With this approach, the user would need to always manually sync text block style and link style.

**Review by JoCa96 — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxLink/types.ts (hunk 1 above)
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxLink/types.ts]: Done

**Review by JoCa96 — APPROVED** [review]

File: packages/sit-onyx/src/components/OnyxLink/OnyxLink.vue [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<script lang="ts" setup>
+import type { OnyxLinkProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxLinkProps>(), {
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxLink/OnyxLink.vue:4]: didn't we want to set href to empty string as default?
image

e.g. in https://github.com/SchwarzIT/onyx/issues/294 the link is only used as click handler without navigating thus the href does not have to be set

**Review by BoppLi — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxLink/OnyxLink.vue (hunk 2 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxLink/OnyxLink.vue:4]: Ah, it's conform with the ticket AC of #237 . guess we'll then need to set the empty href when implementing #294 on that instance.

**Review by BoppLi — DISMISSED** [review]

File: packages/sit-onyx/src/components/OnyxLink/OnyxLink.vue (hunk 2 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxLink/OnyxLink.vue:4]: question goes to @mj-hof

**Review by BoppLi — APPROVED** [review]

**Pull request merged**